apollo-router-core/src/cache.rs

```diff
@@ -0,0 +1,176 @@
+use crate::{CacheResolver, CacheResolverError};
+use futures::lock::Mutex;
+use lru::LruCache;
+use std::cmp::Eq;
+use std::collections::HashMap;
+use std::fmt;
+use std::hash::Hash;
+use tokio::sync::broadcast::{self, Sender};
+
+/// A result from a cache get().
+pub type CacheResult<V> = Result<V, CacheResolverError>;
```

Comment:
Avoid type alias if possible. I'm really not a big fan of them as they obscure the code.

Reply:
I like type aliases to reduce "cognitive burden". Most of the time when I read code, I just want to know what a "type" is representing (in this case a "CacheResult"). If I really want to know what composes a "CacheResult", it's easy enough to find the type alias.

Reply:
Those are the kind of things that makes my life miserable. If you do want to use type aliases like that, let's open up a discussion and let the team vote. But I feel strongly against it. https://twitter.com/CecileTonglet/status/1432931198746206214?s=20

Reply:
this feels like a long discussion, we can have it as a team if you want, but it should not block the merge of this PR. We can always revisit style issues later

Reply:
That's why I said "open up a discussion"... because I didn't want to go into lengthy discussion here...

For now, remove the type alias to make me happy and if you really want to add type alias that I personally feel that obscures the code, open up a discussion on GitHub, let me explain my point better, and let's vote properly. It's not just about "Should we use type aliases in our source code?", that would be really silly.

Reply:
Done in: https://github.com/apollographql/router/pull/235/commits/84a043d61076bfa66746868501519addb17f2193